### PR TITLE
(ISSUE #134) Added super call to onActivityResult in DropInActivity

### PR DESCRIPTION
**Pull request by scannillo**
Fix in response to [Issue #134](https://github.com/braintree/braintree-android-drop-in/issues/134) & [Issue #122](https://github.com/braintree/braintree-android-drop-in/issues/122). 

*Problem:* With the PayPal app installed, the drop in would get stuck on an infinite load after app switch back from PayPal. `onActivityResult` was not invoked. This required merchants to call `           super.onActivityResult(requestCode, resultCode, data)` in their `MainActivity.java`'s `onActivityResult` override method.

*Solution:* Add the `super.onActivityResult(requestCode, resultCode, data)` to the `DropInActivity`'s `onActivityResult` method override so merchant doesn't isn't requires to make this call.

Credits:
@xX7

*Note:* Couldn't think of a meaningful test to add for this super call code addition that doesn't already exist in the tests. Any suggestions appreciated.

### CODE REVIEW

File: Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java [hunk 1]
```diff
@@ -328,6 +328,8 @@ protected void onActivityResult(int requestCode, final int resultCode, Intent da
          * The temporary fix is to forward the data back to BraintreeFragment when we detect
          * the 79129 requestCode.
          */
+        super.onActivityResult(requestCode, resultCode, data);
+
         if (requestCode == 79129) {
```
Comment (quinnjn) [inline Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java]: Since we call `super.onActivityResult`, can this work around be dropped? 

See [Edit2](https://github.com/braintree/braintree-android-drop-in/issues/134#issue-461217287)

File: Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java [hunk 2]
```diff
@@ -328,6 +328,8 @@ protected void onActivityResult(int requestCode, final int resultCode, Intent da
          * The temporary fix is to forward the data back to BraintreeFragment when we detect
          * the 79129 requestCode.
          */
+        super.onActivityResult(requestCode, resultCode, data);
```
Comment (quinnjn) [inline Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java]: The comment above does not seem related to this line

File: Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java (hunk 1 above)
Comment (scannillo) [reply Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java]: Gotcha! My initial thought was that the workaround was still needed as a backup. But I appreciate you pointing it out again. Reading through that a second time makes sense as to why this workaround can now be removed with the super call added. 👍

**Comment by scannillo** [comment]
> DropInActivity#onActivityResult handles add card, and delete payment method results from Drop-in activities. 
> This changes the view state around, i.e. mLoadingViewSwitcher.setDisplayedChild(0);
If we're ending up here for BraintreeFragment#startActivityForResult we might have to do some work to return early before we start changing with Drop-in view states.

I understand what you mean about guarding to return early before changing view states. But am not sure what case would trigger that later scenario. Could you specify a bit?

Thanks for the feedback!

**Comment by xX7** [comment]
Thanks for the Credits!

I think what @quinnjn means is, that the code including and below `mLoadingViewSwitcher.setDisplayedChild(0);` "now" always gets called as soon as a activity started by `BraintreeFragment#startActivityForResult` finishes, which could lead to unintended behavior. This could be fixed/adjusted by checking for `DropInActivity#onActivityResult` desired requestCode's first, before doing any view state switching...

**Comment by scannillo** [comment]
> Is there a test we can write for this behavior? Or not really?
I thought that the already existing tests around view state would be enough to verify that these changes are appropriate ([example test here](https://github.com/braintree/braintree-android-drop-in/blob/c73f04e61f13071a1f59a70109ca04eea4bd915a/Drop-In/src/test/java/com/braintreepayments/api/dropin/DropInActivityUnitTest.java#L603-L611)). 

However, I supposed we could add a test that tests the `super` function is called every time we call `onActivityResult`. I honestly am not sure what would be the most efficient thing to test and would love some suggestions/feedback.

**Comment by scannillo** [comment]
Also, given my research on how `setDisplayedChild(0)` works, it displays the payment selection view of the drop-in.

So it makes sense to me to add it to these 3 situations in the code:
1) RESULT_CANCELED for ADD_CARD_REQUEST_CODE: The user closed the add card flow and we need to show the Drop-in selector again
2) RESULT_OK for ADD_CARD_REQUEST_CODE: The user submitted a credit card and we have a nonce
3) RESULT_OK for DELETE_PAYMENT_METHOD_NONCE_CODE: We successfully deleted a saved payment method

I know @Epreuve and @quinnjn had some concern with this change and I wanted to get their expertise on this as soon as they're free.
